File: src/Log.ts
import { twoDec } from "./Utils";

export enum LogLevel {
    None = 0,
    Error = 1,
    Important = 2,
    Detail = 3,
    Verbose = 4
}

export class Log {
    static setLogLevel(value: LogLevel) {
        Log.level = value;
    }

    static measure(level: LogLevel, arg0: string, action: () => void) {
        if (this.willLog(level)) {
            let start = new Date().getTime()
            action();
            Log.writeLine(level, arg0, twoDec((new Date().getTime() - start) / 1000.0), "sec");
        } else {
            action();
        }
    }

    private static level: LogLevel = LogLevel.Important;

    public static writeLine(level: LogLevel, message?: any, ...args: any) {
        if (this.willLog(level))
            console.log(message, ...args)
    }

    static willLog(level: LogLevel): boolean {
        return level <= Log.level;
    }

}

File: android-sample-app/fast-hierarchical-clustering/src/main/java/com/ganaye/pascal/utils/Log.java
package com.ganaye.pascal.utils;

import static com.ganaye.pascal.utils.Utils.twoDec;

//
public class Log {
    public static LogLevel logLevel = LogLevel.Important;

    //
    public static boolean willLog(LogLevel level) {
        return level.value <= Log.logLevel.value;
    }

    public static void writeLine(LogLevel level, String line) {
        if (willLog(level)) {
            System.out.println(line);
        }
    }


    public static double measure(LogLevel level, String line, Runnable action) {
        double ellapsed = 0;
        long start = System.nanoTime();
        action.run();
        ellapsed = (System.nanoTime() - start) / 1.e9;
        if (willLog(level)) {
            Log.writeLine(level, line + " " + twoDec(ellapsed) + "sec");
        }
        return ellapsed;
    }
//
}